File: services/Reservation.java
package services;

import models.Objet;

public class Reservation {
    private Objet objet;
    private int duree;

    public Reservation(Objet objet, int duree) throws Exception {
        if (!objet.estDisponible()) {
            throw new Exception("L'objet '" + objet.getNom() + "' est déjà réservé.");
        }
        this.objet = objet;
        this.duree = duree;
        objet.reserver();
    }

    public void annuler() {
        objet.liberer();
        System.out.println("Réservation annulée pour " + objet.getNom());
    }

    public void afficher() {
        System.out.println("Réservation de " + objet.getType() + " '" + objet.getNom() + "' pour " + duree + " jour(s).");
    }
}

File: reservation.py
class Reservation:
    def __init__(self, objet, duree):
        if not objet.est_disponible():
            raise Exception(f"L'objet '{objet.nom}' est déjà loué.")
        self.objet = objet
        self.duree = duree
        if not self.objet.reserver():
            raise Exception("Échec de la réservation.")

    def annuler(self):
        self.objet.liberer()
        print(f"Réservation annulée pour {self.objet.nom}")

    def afficher(self):
        print(f"{self.objet.get_type()} '{self.objet.nom}' réservé pour {self.duree} jour(s).")
